akeneo-design-system/src/components/Pagination/Pagination.tsx

```
@@ -0,0 +1,181 @@
+import React, {FC} from 'react';
+import styled, {css} from 'styled-components';
+import {getColor, AkeneoThemedProps} from '../../theme';
+
+type PaginationProps = {
+  /**
+   * The current page number (starting à 1).
+   */
+  currentPage: number;
+
+  /**
+   * The total number of pages.
+   */
+  itemsTotal: number;
+
+  /**
+   * The number of items per page. Default to 25.
+   */
+  itemsPerPage?: number;
+
+  /**
+   * Handler called when a pagination item is clicked.
+   */
+  onClick: (page: number) => void;
+};
+
+const MAX_PAGINATION_ITEMS_WITHOUT_SEPARATOR = 4;
+const PAGINATION_SEPARATOR = '...';
+
+const Pagination: FC<PaginationProps> = ({currentPage, itemsTotal, itemsPerPage = 25, onClick}) => {
+  if (itemsPerPage <= 0) {
+    return <></>;
+  }
+
+  const numberOfPages = Math.ceil(itemsTotal / itemsPerPage);
+
+  if (numberOfPages <= 1 || currentPage > numberOfPages) {
```

Same here: having only one page is possible (if there is less than 25 items) but having a current page above the number of page in total means that there is something wrong and it should throw an error IMHO. What do you think?